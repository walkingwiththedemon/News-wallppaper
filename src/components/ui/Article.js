import { Component } from 'react'
import PropTypes from 'prop-types'
import Like from './Like'
import Comment from './Comment'
import PublicationDate from './PublicationDate'
import { FaTrashO } from 'react-icons/lib/fa'

class Article extends Component {
    render() {
        const { background, title, text, selected, amountLikes, amountComments, timestamp, onRemove, onLike } = this.props
        return (
            <article className="news-list__item article" style={{backgroundColor: background}}>
                <div className="article__delete">
                    <button className="button" onClick={onRemove}>
                        <FaTrashO size={24} />
                    </button>
                </div>
                <div className="article__inner">
                    <div className="article__title">{title}</div>
                    <div className="article__text">{text}</div>
                    <div className="article__footer">
                        <div className="article__footer">
                            <div className="ui-statistic">
                                <div className="ui-statistic__item">
                                    <div className="ui-statistic__items">
                                        <div className="ui-statistic__item">
                                            <Like
                                                selected={selected}
                                                amountLikes={amountLikes}
                                                onClick={() => onLike(selected, amountLikes)}
                                            />
                                        </div>
                                        <div className="ui-statistic__item">
                                            <Comment
                                                comments={amountComments}
                                            />
                                        </div>
                                    </div>
                                </div>
                                <div className="ui-statistic__item ui-statistic__item--align--right">
                                    <PublicationDate timestamp={timestamp} />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </article>
        )
    }
}

Article.propTypes = {
    background: PropTypes.string,
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    selected: PropTypes.bool,
    amountLikes: PropTypes.number,
    amountComments: PropTypes.number,
    onLike: PropTypes.func,
    onRemove: PropTypes.func
}

Article.defaultProps = {
    background: '#4cabb1',
    title: (props, propName) =>
        (typeof props[propName] !== 'string') ?
            new Error('News preview title must be a string') :
            (props[propName].length > 200) ?
                new Error('News preview title is over 200 characters') :
                null,
    text: 'undefined',
    selected: false,
    amountLikes: 0,
    amountComments: 0,
    onRemove: f=>f,
    onLike: f=>f
}

export default Article