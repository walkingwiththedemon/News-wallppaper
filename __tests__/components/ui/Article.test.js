import toJSON from 'enzyme-to-json'
import { compose } from 'redux'
import Article from '../../../src/components/ui/Article'

const { shallow, mount } = Enzyme

describe("<Article /> UI Component", () => {

    const shallowExpect = compose(expect,toJSON,shallow)

    it("Renders correct properties", () =>
        shallowExpect(
            <Article
                background="#262626"
                title="test"
                text="test"
                selected={false}
                amountLikes={25}
                amountComments={1}
                timestamp="01.02.10"
            />
        ).toMatchSnapshot()
    )

    it("Invokes onLike property", () => {
        mount(<Article background="#222222" title="test" text="test" />)
            .find('div.ui-like')
            .childAt(1)
            .simulate('click')
    })

    it("Invokes onRemove property", () => {
        mount(<Article background="#222222" title="test" text="test" />)
            .find("button")
            .simulate('click')
    })

})