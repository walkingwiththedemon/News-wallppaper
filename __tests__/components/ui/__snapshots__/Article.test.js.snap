// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> UI Component Renders correct properties 1`] = `
<article
  className="news-list__item article"
  style={
    Object {
      "backgroundColor": "#262626",
    }
  }
>
  <div
    className="article__delete"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      <FaTrashO
        size={24}
      />
    </button>
  </div>
  <div
    className="article__inner"
  >
    <div
      className="article__title"
    >
      test
    </div>
    <div
      className="article__text"
    >
      test
    </div>
    <div
      className="article__footer"
    >
      <div
        className="article__footer"
      >
        <div
          className="ui-statistic"
        >
          <div
            className="ui-statistic__item"
          >
            <div
              className="ui-statistic__items"
            >
              <div
                className="ui-statistic__item"
              >
                <Like
                  amountLikes={25}
                  onClick={[Function]}
                  selected={false}
                />
              </div>
              <div
                className="ui-statistic__item"
              >
                <Comment
                  comments={1}
                />
              </div>
            </div>
          </div>
          <div
            className="ui-statistic__item ui-statistic__item--align--right"
          >
            <PublicationDate
              timestamp="01.02.10"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</article>
`;
